packages/nextjs/src/app-router/server/auth.ts

```diff
@@ -52,6 +51,9 @@ export const auth = () => {
     noAuthStatusMessage: authAuthHeaderMissing(),
   })(buildRequestLike());
 
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { notFound, redirect } = require('next/navigation');
```

Comment:
`auth()` should never be used in v12, but due to the module structure of the package and the unconditional import of `next/navigation`, in a CJS environment node was trying to resolve the `next/navigation` import. Shifting this inline fixes it.